feat: make message encoding and decoding consistent

nventuro opened the pull request:
We used to have lots of hapzardly put together encoding and decoding functions, now they all live in the same place and follow similar structures and have tests.  We still want to move them (https://github.com/AztecProtocol/aztec-packages/issues/16771), and we still want to review what we call 'content' and what we call 'plaintext', but that'll be a task for later.

These changes very nicely show how events are currently undercommitted and underdecoded, but with them it's very easy to fix these things.

benesjan reviewed (approved):
Perfect!

Happy this ugly tech debt is gone

nventuro committed:
feat: make message encoding and decoding consistent

We used to have lots of hapzardly put together encoding and decoding functions, now they all live in the same place and follow similar structures and have tests.  We still want to move them (https://github.com/AztecProtocol/aztec-packages/issues/16771), and we still want to review what we call 'content' and what we call 'plaintext', but that'll be a task for later.

These changes very nicely show how events are currently undercommitted and underdecoded, but with them it's very easy to fix these things.

benesjan commented:
Fixed the imports here and merged it because I am nice 😇
